src/sumcheck/sumcheck_single.rs

```diff
@@ -72,7 +79,15 @@ where
         SumcheckPolynomial::new(vec![eval_0, eval_1, eval_2], 1)
     }
 
-    /// Compute the polynomial that represents the sum in the first variable.
+    /// Computes the sumcheck polynomial `S(X)`, which is quadratic.
+    ///
+    /// The sumcheck polynomial is given by:
+    ///
+    /// ```text
+    /// S(X) = ∑ p(β) * eq(β)
```

Comment:
Note, here we are not doing eq anymore, we have generic weights. Also, add X to rhs